api/src/main/java/org/apache/flink/agents/api/configuration/ReadableConfiguration.java

```diff
@@ -26,7 +26,16 @@ public interface ReadableConfiguration {
      * @param defaultValue The default value to return if key is not found
      * @return The integer value associated with the key or the default value
      */
-    int getInt(String key, Integer defaultValue);
+    int getInt(String key, int defaultValue);
```

Comment:
Should we use  `Integer` instead of `int` so that we can have a null default value. Same for the other getter. For setter, it is fine to use the `int`, etc.